.github/actionlint.yml

```diff
@@ -0,0 +1,11 @@
+self-hosted-runner:
```

Comment:
Something I just noticed we're missing: all the new files will need copyright headers so that we don't get a follow-up PR from the copyright bot

Reply:
Copyright wasn't a work item on the migration [ticket](https://github.com/hashicorp/GHA-migration/issues/241), which is why it wasn't in scope/addressed. When I run the plan locally (`copywrite headers --plan --spdx MPL-2.0`) it doesn't flag any of the .github files. So this may not be an issue

Reply:
I'd like to suggest maybe that get fixed outside of this PR as the plan appears to flag a large number of other files in the repository. Perhaps as part of an effort to also enable a lint action (we have a .golangci.yml, but no lint workflow in the CI)?

Reply:
Added copyright to several of the files that were being picked up by the cli tool

Reply:
@tgross Do the changes I've mentioned above satisfy this concern?

In short:
1. Add copyright headers to files created here.
2. Introduce a separate PR to add a lint check for the project going forward.

Reply:
Works for me!